### PR TITLE
add xml formatting helper function and integrate into xml serializer

### PR DESCRIPTION
**Pull request by kenkouot**
Adds a function, `formatXml` to the XML serializer.
Also improved the demo and fixed a bug with modifying references on the type classes.

### CODE REVIEW

File: src/serializers/xml/helpers.js [hunk 1]
```diff
@@ -9,3 +9,62 @@ export function equals(leftVal, rightVal, options) {
...
+		let closing = Boolean(ln.match(/<\/.+>/)); // is this a closing tag? ex. </a>
+		let opening = Boolean(ln.match(/<[^!].*>/)); // is this even a tag (that's not <!something>)
+		let type = single ? 'single' : closing ? 'closing' : opening ? 'opening' : 'other';
+		let fromTo = lastType + '->' + type;
+		lastType = type;
+		let padding = '';
+
+		indent += transitions[fromTo];
+
+		for (var j = 0; j < indent; j++) {
```
Comment (lizlux) [inline src/serializers/xml/helpers.js:55]: `let`

File: src/serializers/xml/helpers.js [hunk 2]
```diff
@@ -9,3 +9,62 @@ export function equals(leftVal, rightVal, options) {
 		return options.fn(this);
 	}
 }
+
+// Implementation from http://stackoverflow.com/a/2893259
+export function formatXml(xml) {
```
Comment (lizlux) [inline src/serializers/xml/helpers.js:14]: It's a little funny that a handlebars helper and a non handlebars helper are in the same file. Not a big deal just has the potential to be misleading.

File: src/serializers/xml/template.js [hunk 3]
```diff
@@ -1,43 +1,37 @@
...
 	{{#defaultValue}}<default>{{.}}</default>{{/defaultValue}}
 	{{#stringTemplate}}<format>{{.}}</format>{{/stringTemplate}}
 </data>`;
 
-const image = `
-<image source="{{boundVariableName}}">
+const image = `<image{{#boundVariableName}} source="{{.}}"{{/boundVariableName}}>
 	{{#caption}}
-		<caption source="{{boundVariableName}}">
+		<caption{{#boundVariableName}} source="{{.}}"{{/boundVariableName}}>
```
Comment (lizlux) [inline src/serializers/xml/template.js:9]: Isn't `boundVariableName` required for a caption to be made editable?

File: src/serializers/xml/template.js [hunk 4]
```diff
@@ -1,43 +1,37 @@
...
 			{{#defaultValue}}<default>{{.}}</default>{{/defaultValue}}
 			{{#stringTemplate}}<format>{{.}}</format>{{/stringTemplate}}
 		</caption>
 	{{/caption}}
 	{{#if altDefaultValue}}
-		<alt source="{{#if altBoundVariableName}}{{altBoundVariableName}}{{else}}alternative_title{{/if}}">
+		<alt{{#altBoundVariableName}} source="{{.}}"{{/altBoundVariableName}}>
```
Comment (lizlux) [inline src/serializers/xml/template.js:15]: same question with alt tags

File: src/serializers/xml/template.js (hunk 3 above)
Comment (kenkouot) [inline src/serializers/xml/template.js:9]: I don't understand the question. Can you elaborate?

Comment (lizlux) [inline src/serializers/xml/template.js:9]: discussed offline.

File: src/serializers/xml/template.js (hunk 4 above)
Comment (kenkouot) [inline src/serializers/xml/template.js:15]: discussed offline.

**Pull request merged**